components/gitpod-cli/cmd/tasks-stop.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	supervisor_helper "github.com/gitpod-io/gitpod/gitpod-cli/pkg/supervisor-helper"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	supervisor "github.com/gitpod-io/gitpod/supervisor/api"
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+)
+
+var stopTaskCmdOpts struct {
+	All bool
+}
+
+// stopTaskCmd represents the stop task command
+var stopTaskCmd = &cobra.Command{
+	Use:   "stop <id>",
+	Short: "Stop a workspace task",
+	Args:  cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		var (
+			terminalAliases []string
+			ctx             context.Context
+			cancel          context.CancelFunc
+		)
+
+		terminalClient, err := supervisor_helper.GetTerminalServiceClient(context.Background())
+
+		if err != nil {
+			log.Fatalf("cannot get terminal service: %s", err)
+			return
+		}
+
+		ctx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		all, _ := cmd.Flags().GetBool("all")
+
+		if all {
+			ctx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+
+			tasks, err := supervisor_helper.GetTasksListByState(ctx, supervisor.TaskState_running)
+
+			if err != nil {
+				log.Fatalf("cannot get task list: %s", err)
+				return
+			}
+
+			if len(tasks) == 0 {
+				fmt.Println("There are no running tasks")
+				return
+			}
+
+			for _, task := range tasks {
+				terminalAliases = append(terminalAliases, task.Terminal)
+			}
+		} else if len(args) > 0 {
+			_, err := terminalClient.Get(ctx, &api.GetTerminalRequest{
+				Alias: args[0],
+			})
+
+			if err != nil {
+				fmt.Printf("The selected task was not found: %s.\nMake sure to use the correct task ID.\nUse 'gp tasks list' to obtain the task id or run 'gp tasks stop' to select the desired task\n", args[0])
```

Comment:
Is it too verbose, what do you guys think?

I like the idea from @filiptronicek to provide more info because the task ID could be confusing for some users.

<img width="781" alt="Screenshot 2022-08-21 at 19 43 21" src="https://user-images.githubusercontent.com/2318450/185806320-cc54d663-0072-4982-b3e5-570977415af4.png">

Reply:
I like it this way, I think it makes for a message that users without an idea about task IDs can understand.